### ISSUE TITLE
Automatic file extension for session files and workspace files

### ISSUE THREAD

**Issue by Uhf7** [notepad-plus-plus/notepad-plus-plus#8311]
In _Preferences --> MISC_, two special file extensions can be specified. One for session files, and another one for workspace files. There is another setting in _Preferences --> Default directory_, named Use _new style dialog ..._.

When I save a new session file or a new workspace file, then the behavior of the **Save As** dialog is unnecessarily dependent of whether I use the _new style_ dialog or not. Using the standard **Save As** dialog, the extension of the session/workspace file is automatically appended, if specified in _Preferences --> MISC_. Using the _new style_ dialog, it is not.

I understand, that not automatically appending the extension is the designed behavior of the _new style_ dialog. This may be good for the user's document files, but it seems not to be so good for session and workspace files.

As soon as I specify an extension for session files or workspace files in _Preferences_, I modify the way in which these files are seen by Npp anyway, and from that point on, it is quite useless for me to store those files without their assigned extensions.

So, long story short, I want to append the extensions for session files and workspace files automatically in the _new style_ **Save As** dialog too, and only, if the extensions are specified in _Preferences_. That would it make easier to save new session or workspace files with the correct extension. (I prefer the _new style_ dialog for the list with my hard drives on the left side instead of the big icons.)

To facilitate the usability for the many users who are already used to type in those extensions manually, or select an existing file from the file list, it should be implemented in the same way as the Windows Notepad appends the **.txt** extension automatically: This extension is only appended, if it is not already there.

**Comment by donho** [comment]
@Uhf7 
The inconvenient old style save dialog should be removed. It's kept because file extension adding feature and Unix style path capacity could not implemented in the new style dialog.
Now file extension adding feature is added by your PR. If Unix style path capacity can be done in the new style dialog, then the old style will be removed (no COM Object implementation) - that will improve user XP and make the code simpler.
